source/module_pw/pw_basis.cpp

```diff
@@ -10,46 +10,54 @@ namespace ModulePW
 
 PW_Basis::PW_Basis()
 {
-    ig2isz = NULL;
-    istot2bigixy = NULL;   
-    ixy2istot = NULL;
-    is2ixy = NULL;
-    ixy2ip = NULL; 
-    startnsz_per = NULL;
-    nstnz_per = NULL;
-    nst_per = NULL;
-    gdirect = NULL;		
-    gcar = NULL; 
-    gg = NULL;
-    startz = NULL;
-    numz = NULL;  
-    this->numg = NULL;
-	this->startg = NULL;
-	this->startr = NULL;
-	this->numr = NULL;
+    ig2isz = new int [1];
+    istot2bigixy = new int [1];   
+    ixy2istot = new int [1];
+    is2ixy = new int [1];
+    ixy2ip = new int [1];
+    startnsz_per =new int [1];
+    nstnz_per = new int [1];
+    nst_per = new int [1];
+    gdirect = new ModuleBase::Vector3<double> [1];		
+    gcar = new ModuleBase::Vector3<double> [1]; 
+    gg = new double [1];
+    startz = new int [1];
+    numz = new int [1]; 
+    numg = new int [1];
+	startg = new int [1];
+	startr = new int [1];
+	numr = new int [1];
+    ig2igg = new int [1];
+    gg_uniq = new double [1];
+    ig_gge0 = -1;
```

Comment:
Because vc-relax calculation should re-initialize members. If I use former coedes, I need to write a clean function before re-initializing. The new codes also passed valgrind memory test of the test_gnu.sh script in module_pw/test.

Reply:
Maybe you can use "if(ig2isz != NULL) delete[] ig2isz;" to handle the re-initialize

Reply:
It cannot handle it.  The leak of memory comes from, for example, "ig2isz=new int [20]". When I don not delete[] ig2isz and reinitialize. 20-int data will lose and cause leak of memory. Thus another solution is to write a function of clean. However, new int [1] is also not bad solution and do not to call clean function before re-initialize.

Reply:
> Maybe you can use "if(ig2isz != NULL) delete[] ig2isz;" to handle the re-initialize

@Qianruipku You can just delete a null pointer, the int[1] placeholder is not necessary at all.